File: rpg-backend/app/schemas/quest.py
from pydantic import BaseModel

from app.schemas.common import HeroProgressOut


class QuestOut(BaseModel):
    id: int
    code: str
    name: str
    description: str | None
    condition_type: str
    target_value: int
    current_progress: int
    is_completed: bool
    is_claimed: bool
    is_active_slot: bool
    reward_xp: int
    reward_coins: int


class QuestClaimOut(BaseModel):
    id: int
    code: str
    name: str
    reward_xp: int
    reward_coins: int
    claimed_at: str
    hero_progress: HeroProgressOut

File: rpg-frontend/src/types/quest.ts
import type { HeroProgressOut } from "./common";

export interface QuestOut {
  id: number;
  code: string;
  name: string;
  description: string | null;
  condition_type: string;
  target_value: number;
  current_progress: number;
  is_completed: boolean;
  is_claimed: boolean;
  is_active_slot: boolean;
  reward_xp: number;
  reward_coins: number;
}

export interface QuestClaimOut {
  id: number;
  code: string;
  name: string;
  reward_xp: number;
  reward_coins: number;
  claimed_at: string;
  hero_progress: HeroProgressOut;
}
